### PR TITLE
[3.9] bpo-40939: Fix test_keyword for the old parser

### PR DESCRIPTION
**Pull request by pablogsal**
https://bugs.python.org/issue40939

### CODE REVIEW

**Comment by lysnikolaou** [comment]
You beat me to it!! Should we maybe also rename the easter egg in 3.9 as per @vstinner's suggestion?

**Comment by pablogsal** [comment]
> You beat me to it!!

Remember: fastest gun in the west 🔫 🤠 

> Should we maybe also rename the easter egg in 3.9 as per @vstinner's suggestion?

¯\_(ツ)_/¯ As you prefer: there is no possibility that it needs to be future-proof now :P

**Comment by lysnikolaou** [comment]
> Remember: fastest gun in the west 🔫 🤠

Time and time again!

> Should we maybe also rename the easter egg in 3.9 as per @vstinner's suggestion?

You both know best.

**Comment by pablogsal** [comment]
> You both know best.

Ok, will change it here then

File: Lib/test/test_keyword.py [hunk 1]
```diff
@@ -21,7 +22,10 @@ def test_changing_the_kwlist_does_not_affect_iskeyword(self):
         self.assertFalse(keyword.iskeyword('eggs'))
 
     def test_all_keywords_fail_to_be_used_as_names(self):
-        for key in keyword.kwlist:
+        all_keywords = set(keyword.kwlist)
+        if use_old_parser():
+            all_keywords -= {'__new_parser__'}
```
Comment (vstinner) [inline Lib/test/test_keyword.py]: I would use discard(), but your code is correct ;-)

```suggestion
            all_keywords.discard('__new_parser__')
```

Comment (pablogsal) [reply Lib/test/test_keyword.py]: > I would use discard(), but your code is correct ;-)

I like `discard()` :)

**Comment by pablogsal** [comment]
The buildbot failure is unrelated

**Pull request merged**